### PR TITLE
Added putAll(Collection) and putAll(Array) methods.

### PR DESCRIPTION
**Pull request by ethauvin**
I've added two methods to add/put all of the elements of an array/collection to the `JSONArray`.

Mostly to avoid inefficient code like:

```java
JSONArray jsonArray;
...
if (myArray != null) {
    jsonArray = new JSONArray(myArray);
} else {
    jsonArray = new JSONArray();
}
```
You can now do:
```java
JSONArray jsonArray = new JSONArray();
...
if (myArray != null && !myArray.isEmpty()) {
    jsonArray.putAll(myArray);
}
```
It is also very useful if you need to add elements from multiple arrays/collections. For example:
```java
int[] myInts = { 1, 2, 3, 4, 5 };
List<String> myList = Arrays.asList("one", "two", "three", "four", "five");
```
Instead of:
```java
JSONArray jsonArray = new JSONArray(myInts);

for (String s : myList) {
    jsonArray.put(s);
}
```
Simply do:
```java
JSONArray jsonArray = new JSONArray(myInts);
jsonArray.putAll(myList);
```

There's no real code change, just things moved around. I did not implement tests since the constructors are already tested.

Let me know if you have any questions.

### CODE REVIEW

**Comment by stleary** [comment]
Thanks for the pull request, but it can't be accepted at the present time. Please see [How do you decide which pull requests to accept?](https://github.com/stleary/JSON-java/wiki/FAQ#how-do-you-decide-which-pull-requests-to-accept)

**Comment by viveksacademia4git** [comment]
@ethauvin I wanted to add something similar in issue #528; proposed by you in this issue.

**Comment by ethauvin** [comment]
@viveksacademia4git You sure did. It's a pretty obvious design flaw for a Java library.

**Comment by stleary** [comment]
At the end of the day, this is an uncomplicated reference library with the limited goal of demonstrating how to parse JSON in Java. Being relatively fast and small helps too, I suppose. Currently, this project is in maintenance mode. The main criteria for accepting PRs are to fix bugs and address major user inconveniences (please see the [FAQ section](https://github.com/stleary/JSON-java/wiki/FAQ#how-do-you-decide-which-pull-requests-to-accept)). With regard to #529 and #528, neither PR made a compelling case for inclusion, nor did others raise comments to support the need for this change.

**Comment by ethauvin** [comment]
@stleary I completely understand the current motivations. Although as far as #529 is concerned, since it was closed within hours of submission, I don't see how anyone would support it. It would have been interesting to see what others had to say.

**Comment by stleary** [comment]
No objection to reopening the PR for a few days.

File: src/main/java/org/json/JSONArray.java [hunk 1]
```diff
@@ -1500,6 +1519,44 @@ public Writer write(Writer writer, int indentFactor, int indent)
...
+     *
+     * @throws JSONException
+     *            If not an array or if an array value is non-finite number.
+     * @throws NullPointerException
+     *            Thrown if the array parameter is null.
+     */
+    private void addAll(Object array) throws JSONException {
+        if (array.getClass().isArray()) {
+            int length = Array.getLength(array);
+            this.myArrayList.ensureCapacity(length);
```
Comment (johnjaylward) [inline src/main/java/org/json/JSONArray.java:1551]: this.myArrayList.ensureCapacity(this.myArrayList + length);

File: src/main/java/org/json/JSONArray.java [hunk 2]
```diff
@@ -1500,6 +1519,44 @@ public Writer write(Writer writer, int indentFactor, int indent)
...
+
+
+    /**
+     * Add a collection's elements to the JSONArray.
+     *
+     * @param collection
+     *            A Collection.
+     */
+    private void addAll(Collection<?> collection) {
+        for (Object o: collection){
```
Comment (johnjaylward) [inline src/main/java/org/json/JSONArray.java:1531]: Add a new line:
```java
this.myArrayList.ensureCapacity(this.myArrayList.size()+ collection.size());
```

Comment (ethauvin) [reply src/main/java/org/json/JSONArray.java:1531]: Implemented in 86e136a

Comment (johnjaylward) [reply src/main/java/org/json/JSONArray.java:1531]: I think you missed this one. You got the one for the `addAll(Array)`, but not for `addAll(Collection)`

**Comment by ethauvin** [comment]
@johnjaylward Duh! Thanks!

Fixed in f1d354c

**Comment by stleary** [comment]
Please add unit tests that explicitly call the new putAll() methods.
Please add an example in the description for, "It is also very useful if you need to add elements from multiple arrays/collections."

**Comment by ethauvin** [comment]
@stleary See 0d13e56

**Comment by stleary** [comment]
**What problem does this code solve?**
Added new methods to JSONArray to make it more convenient to add arrays and collections all at once,
instead of a single element at a time.
This is not a bug fix.
Unit tests were added to exercise the new API methods.

**Risks**
Low

**Changes to the API?**
New putAll() methods were added to JSONArray
New private addAll() methods were added to implement the functionality.

**Will this require a new release?**
No

**Should the documentation be updated?**
No

**Does it break the unit tests?**
No

**Was any code refactored in this commit?**
The JSONArray array and collection constructors were updated to use new private methods addAll()

**Review status**
**APPROVED**

**Comment by stleary** [comment]
Starting 3-day comment window.

**Pull request merged**